File: server/internal/bills/bill_payment.go
package bills

import (
	"math/big"
	"time"

	"gorm.io/gorm"
)

type BillPayment struct {
	ID         uint           `json:"id" gorm:"primaryKey"`
	BillTypeID uint           `json:"bill_type_id" gorm:"not null;index"`
	Year       int            `json:"year" gorm:"not null"`                      // format: YYYY
	Month      int            `json:"month" gorm:"not null"`                     // format: MM
	Amount     string         `json:"amount" gorm:"type:decimal(10,2);not null"` // Store as string for precision
	Note       string         `json:"note"`
	UserID     uint           `json:"user_id" gorm:"not null;index"`
	CreatedAt  time.Time      `json:"created_at"`
	UpdatedAt  time.Time      `json:"updated_at"`
	DeletedAt  gorm.DeletedAt `json:"-" gorm:"index"`

	// Associations
	BillType BillType `json:"bill_type,omitempty" gorm:"foreignKey:BillTypeID"`
}

// GetAmountAsBigFloat converts the stored string amount to big.Float
func (bp *BillPayment) GetAmountAsBigFloat() *big.Float {
	amount := new(big.Float)
	amount.SetString(bp.Amount)
	return amount
}

// SetAmountFromBigFloat sets the amount from a big.Float
func (bp *BillPayment) SetAmountFromBigFloat(amount *big.Float) {
	bp.Amount = amount.String()
}

File: ios/backup/JiceotApp/Models/BillPayment.swift
//
//  BillPayment.swift
//  JiceotApp
//
//  Created on 12/11/2025.
//

import Foundation

struct BillPayment: Codable, Identifiable {
    let id: Int
    let billTypeId: Int
    let year: Int
    let month: Int
    let amount: String
    let note: String
    let userId: Int
    let createdAt: String
    let updatedAt: String
    let billType: BillType?
    
    enum CodingKeys: String, CodingKey {
        case id
        case billTypeId = "bill_type_id"
        case year
        case month
        case amount
        case note
        case userId = "user_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case billType = "bill_type"
    }
}

struct CreateBillPaymentRequest: Codable {
    let billTypeId: Int
    let year: Int
    let month: Int
    let amount: String
    let note: String?
    
    enum CodingKeys: String, CodingKey {
        case billTypeId = "bill_type_id"
        case year
        case month
        case amount
        case note
    }
}

struct BillPaymentListResponse: Codable {
    let billPayments: [BillPayment]
    let total: Int
    
    enum CodingKeys: String, CodingKey {
        case billPayments = "bill_payments"
        case total
    }
}
